Issue 3

apavlov1992 opened the pull request:
Issue_3: 
- add goroutine semaphore;
- remove several cli arguments; 
- add check existing ID in DB;

Commit:
Issue_3: Without goroutine

Commit:
Issue_3: Without goroutine

Commit:
Issue_3: Without goroutine

Commit:
Issue_3:
- add goroutine semaphore;
- remove several cli arguments;
- add check existing ID in DB;

Commit:
Issue_3:
- add goroutine semaphore;
- remove several cli arguments;
- add check existing ID in DB;

vbelogrudov commented:
build/xkcd: ## Build binary file for xkcd
	env GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o xkcd internal/xkcd/xkcd.go 

надо удалить, а по умолчанию оставить build

vbelogrudov commented:
➜  apavlov1992/golang_course git:(issue_3) go build -o xkcd cmd/main.go 
# command-line-arguments
cmd/main.go:36:12: undefined: NewComicsWorker

vbelogrudov commented on cmd/main.go:
```diff
@@ -15,10 +15,9 @@ import (
-	flag.StringVar(&configFileName, "config", "../config/config.yaml", "Specify configuration file name to use.")
-	numberOfComics := flag.Int("n", 0, "Number of comics to fetch")
-	outputJson := flag.Bool("o", false, "Output JSON in CLI or write to file")
+	flag.StringVar(&configFileName, "config", "/Users/a.pavlov/GolandProjects/golang_course/config/config.yaml", "Specify configuration file name to use.")
```
путь по умолчанию желательно ./config.yaml

apavlov1992 replied:
Да, забыл поменять

apavlov1992 replied:
Done

vbelogrudov reviewed (commented):
прекрасный пайплайн, осталось заставить ег компилироваться

Commit:
Issue_3:
- add goroutine semaphore;
- remove several cli arguments;
- add check existing ID in DB;

vbelogrudov reviewed (commented):
```
➜  apavlov1992/golang_course git:(issue_3) make
Makefile:11: warning: overriding recipe for target 'build'
Makefile:5: warning: ignoring old recipe for target 'build'
env GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o main cmd/main.go
go: no Go source files
make: *** [Makefile:11: build] Error 1
➜  apavlov1992/golang_course git:(issue_3) cat Makefile 
# Build targets
.PHONY: build/xkcd build/stemming

build: ## Build binary file for xkcd
	env GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o xkcd internal/xkcd/xkcd.go

build/stemming: ## Build binary file for stemming
	env GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o stemming internal/stemming/stemming.go

build: ## Build main projects
	env GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o main cmd/main.go


```

Commit:
Issue_3:
- add goroutine semaphore;
- remove several cli arguments;
- add check existing ID in DB;

vbelogrudov reviewed (approved):
красивый pipeline (fan out-in)!

vbelogrudov commented on cmd/main.go:
```diff
@@ -3,10 +3,11 @@ package main
 import "C"
```
import "C" все портит :)

apavlov1992 replied:
fix

Commit:
Issue_3:
- add goroutine semaphore;
- remove several cli arguments;
- add check existing ID in DB;

vbelogrudov commented on cmd/main.go:
```diff
@@ -29,29 +29,37 @@ func main() {
+	for _, c := range comics {
+		bar.Add(1)
+		if c.Num != 0 {
+			c.Description = strings.Join(stemming.StemmingString(c.Description), " ")
```
лучше оставить слайсом, иначе потом опять разделят ь на слова
